spec/requests/todos_spec.rb

```diff
@@ -5,34 +5,87 @@
   # 正常系
   describe '正常系' do
     describe 'GET #show' do
+      let!(:todo) { create(:todo) }
+
+      before { get "/api/todos/#{todo.id}" }
+
       context '存在するTODOにアクセスしたとき' do
-        it 'TODOの詳細情報を取得できること' do
+        it 'responseのステータスが200であること' do
+          expect(response).to have_http_status(200)
+        end
+        it 'responseの中身が正しいこと' do
+          json = JSON.parse(response.body)
+          expect(json['id']).to eq todo.id
+          expect(json['text']).to eq todo.text
+          expect(json['created_at']).not_to be nil
+          expect(json['updated_at']).not_to be nil
         end
       end
     end
 
     describe 'GET #index' do
-      it 'TODOの一覧を取得できること' do
+      let!(:todo) { create(:todo) }
+
+      before { get "/api/todos" }
+
+      context 'TODOが作成されているとき' do
+        it 'responseのステータスが200であること' do
+          expect(response).to have_http_status(200)
+        end
+        it 'responseの中身が正しいこと' do
+          json = JSON.parse(response.body)
+          expect(json.count).to eq 1
+          expect(json[0]['id']).to eq todo.id
+          expect(json[0]['text']).to eq todo.text
+          expect(json[0]['created_at']).not_to be nil
+          expect(json[0]['updated_at']).not_to be nil
+        end
       end
     end
 
-    describe 'GET #edit' do
+    describe 'PUT #update' do
+      let!(:todo) { create(:todo) }
+
       context '存在するTODOを編集したとき' do
-        it 'TODOが編集できること' do
+        it 'responseのステータスが200であること' do
+          expect do
+            patch api_todo_path(todo.id),
+            params: { todo: {text: "TODOを編集しました"} }
+          end.to change { Todo.find(1).text }
+          expect(response).to have_http_status(200)
+        end
+        it 'responseの中身が正しいこと' do
+          expect do
+            put "/api/todos/#{todo.id}",
+            params: { todo: {text: "TODOを編集しました"} }
+          end.to change { Todo.find(1).text }
+          expect(Todo.find(1).id).to eq todo.id
+          expect(Todo.find(1).created_at).not_to be nil
+          expect(Todo.find(1).updated_at).not_to be nil
         end
       end
     end
 
     describe 'POST #create' do
-      context 'todoのtextが入力されてるとき' do
-        it 'TODOが作成できること' do
-        end
+      let!(:todo) { create(:todo) }
```

Comment:
```suggestion
      let!(:todo) { build(:todo) }
```

createしてしまうとデータベースに保存されてしまい、
それをcreateでpostすると同じ内容のデータでidが違うデータが作成されてしまうので、createではなくbuildの方がテストとしては正しいと思います。

Reply:
そうですよね、id違いでレコードが出来ますよね。
事前にレコード作成する必要がなければ、そもそも
let!(:todo) { create(:todo) } 自体が無くてもいい感じですかね。

buildで書き換えて、byebugでテストを止めてレコードを確認すると
レコードは0でしたので、無くても良さそうと思い、下記のようにしてみたのですが認識大丈夫でしょうか。
```
     describe 'POST #create' do
-      let!(:todo) { create(:todo) }
-
       it 'TODOが作成できること' do
         expect do
           post "/api/todos", params: { todo: attributes_for(:todo) }
```